### ISSUE TITLE
Support Python 3.13

### ISSUE THREAD

**Issue by murrple-1** [pemistahl/lingua-py#239]
No rush from me personally, but during a routine "see if my project can run in Python 3.13" (which was officially released very shortly ago), this project was one of my few dependencies that didn't have a 3.13 version available.

```
ERROR: Ignored the following versions that require a different python version: 0.10.6 Requires-Python >=2.7, <3.6; 0.10.7 Requires-Python >=2.7, <3.6; 1.3.5 Requires-Python <3.13,>=3.8
59.85 ERROR: Could not find a version that satisfies the requirement lingua-language-detector==2.0.2 (from versions: 1.0.0, 1.0.1, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.2.0, 1.2.1, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.3.4)
ERROR: No matching distribution found for lingua-language-detector==2.0.2
```

So this issue is a request/reminder to add a build for Python 3.13.

**Comment by pemistahl** [comment]
Thank you for the reminder. Actually, I read yesterday that Python 3.13 was released. I plan to make new releases of my library with new features this month, both for Python and Rust.  I will make sure that Python 3.13 is supported.

**Comment by JanEricNitschke** [comment]
Hi, id like to come back to this.

I see on [PyPI](https://pypi.org/project/lingua-language-detector/) that there is a 2.0.2 release which supposedly supports python 3.13 (`Requires: Python >=3.8`). However there are no wheels provided for 3.13 and so `pip install lingua-language-detector==2.0.2` fails with `ERROR: No matching distribution found for lingua-language-detector==2.0.2`.

**Comment by pemistahl** [comment]
@JanEricNitschke Please use Python 3.12 for now if you want to use the Rust implementation of my library. Last month, I released Lingua 1.4.0 which supports Python 3.13 and has new features. It's pure Python so it's slower than the Rust implementation. 

Your help is appreciated but I want to implement the new features from the pure Python implementation first before I make a new release with Python 3.13 support. So there is nothing to do for you at the moment. The wheels will be created in CI when I make the next release. My spare time is currently limited but I will make a new release. I just can't tell you when.

**Comment by JanEricNitschke** [comment]
@pemistahl Thanks for the quick reply. 

If i understood correctly then 1.X is the pure python version and 2.X the rust version, right?

Unfortunately it is pretty important for us to use both the rust implementation and make python 3.13 available. Would there be an option to do like a patch release to just get 3.13 support out?

**Comment by pemistahl** [comment]
No, I'm sorry. Each release takes a significant amount of storage space on PyPI which is limited. I won't make a new release only for Python 3.13 support.

**Comment by arthur-st** [comment]
Could there be, by chance, an option to delete the current version supporting `<=3.12` and release a drop-in replacement supporting `<=3.13`? I feel that it would be deviating from the norm for Python packages, to block projects using version 2 of this package from upgrading to Python 3.13 until a sufficiently large new feature release becomes available.

**Comment by pemistahl** [comment]
@arthur-st No, I won't do that. Removing released versions from PyPI is like a punch in the faces of people who already rely on that specific version in their projects because their builds will break. I did that in the past because of significant bug fixes and because of limited storage on PyPI. However, storage on PyPI has luckily been increased for my project so that I don't need to remove released versions anymore.

I'm wondering why you need Python 3.13 so urgently in the first place. Is it because of the new free-threaded mode of operation? Otherwise, I don't see what you can accomplish with 3.13 that you cannot still accomplish with 3.12 already.

**Comment by pemistahl** [comment]
> As to why I want 3.13 support - the first maintenance release is out, so we're starting to use it both for new projects and for maintenance upgrades.

Well, this is no urgent reason, isn't it? Python 3.12 will also get maintenance releases until the end of 2028.

After all issues in [milestone 6](https://github.com/pemistahl/lingua-py/milestone/6) and [milestone 9](https://github.com/pemistahl/lingua-py/milestone/9) have been implemented, I will make a new release. Not earlier. I'm glad that my library is used in real projects. But I'm doing all of this free of charge in my spare time which has never been as limited as it is now. That's why the current progress is slow. That's not nice but I can't help it. I hope you understand.

**Comment by arthur-st** [comment]
I do understand that, and I appreciate your effort as an OSS volunteer. That said, I would like to note that the understanding goes both ways, in that you're not necessarily the person to judge whether if our projects are being prematurely upgraded to 3.13 or not, or how our technical debt planning should look like.

**Comment by pemistahl** [comment]
While waiting for the release, you can compile a Python 3.13 wheel yourself and install it. Maybe this is a temporary solution for you?

You can do it like this:

```
git clone git@github.com:pemistahl/lingua-rs.git
cd lingua-rs
python3 -m venv .venv
./.venv/bin/activate
pip install maturin
maturin build --release
```

You will then find the wheel file in folder `./target/wheels`.

**Comment by arthur-st** [comment]
I'm afraid that won't work for my current projects, but I appreciate the thought.

**Comment by pemistahl** [comment]
Python 3.13 support has been implemented in the Rust implementation, so this issue can be closed.

**Comment by JanEricNitschke** [comment]
> Python 3.13 support has been implemented in the Rust implementation, so this issue can be closed.

I guess to use it i have to install directly from github?

**Comment by arthur-st** [comment]
@JanEricNitschke Seems so. There's been no new release on PyPI when I checked after this was marked as completed, and the PyPI release that's there did fail to install into a Python 3.13 environment, just as before. I'm guessing that what the maintainer may have meant is that Python 3.13 has been incorporated into an unspecified future release of the package.
